cv32e40p/bsp/handlers.S

```diff
@@ -66,158 +66,174 @@
 
 /* exception handling */
 __no_irq_handler:
-	la a0, no_exception_handler_msg
-	jal ra, puts
-	j __no_irq_handler
+  la a0, no_exception_handler_msg
+  jal ra, puts
+  j __no_irq_handler
 
 m_software_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_timer_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_external_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_fast0_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_fast1_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_fast2_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_fast3_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_fast4_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_fast5_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_fast6_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_fast7_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_fast8_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_fast9_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_fast10_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_fast11_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_fast12_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_fast13_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_fast14_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 m_fast15_irq_handler:
-	j __no_irq_handler
+  j __no_irq_handler
 
 u_sw_irq_handler:
-	/* While we are still using puts in handlers, save all caller saved
-	   regs.  Eventually, some of these saves could be deferred.  */
-	addi sp,sp,-64
-	sw ra, 0(sp)
-	sw a0, 4(sp)
-	sw a1, 8(sp)
-	sw a2, 12(sp)
-	sw a3, 16(sp)
-	sw a4, 20(sp)
-	sw a5, 24(sp)
-	sw a6, 28(sp)
-	sw a7, 32(sp)
-	sw t0, 36(sp)
-	sw t1, 40(sp)
-	sw t2, 44(sp)
-	sw t3, 48(sp)
-	sw t4, 52(sp)
-	sw t5, 56(sp)
-	sw t6, 60(sp)
-	csrr t0, mcause
-	li t1, EXCEPTION_ILLEGAL_INSN
-	beq t0, t1, handle_illegal_insn
-	li t1, EXCEPTION_ECALL_M
-	beq t0, t1, handle_ecall
-	li t1, EXCEPTION_BREAKPOINT
-	beq t0, t1, handle_ebreak
-	j handle_unknown
+  /* While we are still using puts in handlers, save all caller saved
+     regs.  Eventually, some of these saves could be deferred.  */
+  addi sp,sp,-76
```

Comment:
Will this offset (-76) still work properly in all cases if `PULP` is not defined?

Reply:
Yes you reserve a stack frame larger than need. There will be few words not used in case of v1.
No problem for that the stack is large enough...